src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeploymentParameters.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting.IIS
+{
+    public class IISDeploymentParameters : DeploymentParameters
+    {
+        public IISDeploymentParameters() : base()
+        {
+            WebConfigActionList = CreateDefaultWebConfigActionList();
+        }
+
+        public IISDeploymentParameters(TestVariant variant)
+            : base(variant)
+        {
+            WebConfigActionList = CreateDefaultWebConfigActionList();
+        }
+
+        public IISDeploymentParameters(
+           string applicationPath,
+           ServerType serverType,
+           RuntimeFlavor runtimeFlavor,
+           RuntimeArchitecture runtimeArchitecture)
+            : base(applicationPath, serverType, runtimeFlavor, runtimeArchitecture)
+        {
+            WebConfigActionList = CreateDefaultWebConfigActionList();
+        }
+
+        public IISDeploymentParameters(DeploymentParameters parameters)
+        {
+            foreach (var propertyInfo in typeof(IISDeploymentParameters).GetProperties())
+            {
+                propertyInfo.SetValue(this, propertyInfo.GetValue(parameters));
```

Comment:
Not all of them are settable, which tells me you've never run this code path.
https://github.com/aspnet/Hosting/blob/8836de411408257dd75f5db5f7d6724bfb0fab9e/src/Microsoft.AspNetCore.Server.IntegrationTesting/Common/DeploymentParameters.cs#L157-L162

Reply:
It would be better to add a copy constructor to DeploymentParameters that explicitly set all of the properties and then call base from here.